app/src/main/java/app/revanced/integrations/youtube/patches/spoof/SpoofAppVersionPatch.java

```diff
@@ -1,12 +1,26 @@
 package app.revanced.integrations.youtube.patches.spoof;
 
+import app.revanced.integrations.shared.Logger;
 import app.revanced.integrations.youtube.settings.Settings;
 
 @SuppressWarnings("unused")
 public class SpoofAppVersionPatch {
 
-    private static final boolean SPOOF_APP_VERSION_ENABLED = Settings.SPOOF_APP_VERSION.get();
-    private static final String SPOOF_APP_VERSION_TARGET = Settings.SPOOF_APP_VERSION_TARGET.get();
+    private static final boolean SPOOF_APP_VERSION_ENABLED;
+    private static final String SPOOF_APP_VERSION_TARGET;
+
+    static {
+        // TODO: remove this migration code
+        // Spoof targets 16.x and 17.x that no longer reliably work.
+        if (Settings.SPOOF_APP_VERSION_TARGET.get().compareTo("18.01.01") < 0) {
```

Comment:
If one set the version to a lower version, I think resetting app data was necessary to restore to newer versions wasn't it?